manifests: Split baseos and extensions builds into separate tasks

LorbusChris opened the pull request:
/cc @sherine-k 

Currently running this at https://console-openshift-console.apps.smaug.na.operate-first.cloud/k8s/ns/okd-team/tekton.dev~v1beta1~PipelineRun/okd-coreos-all-xw0kbz/logs
/hold

sherine-k reviewed (approved):
Great work @LorbusChris .
Thanks!

LorbusChris commented:
Running into a problem with image pushing now. Seems there's a new annotation needed on the secret. Running again with that change (also added to the README)

LorbusChris committed:
manifests: Split baseos and extensions builds into separate tasks

Also:
- moves credentials from service account to workspace
- removes the serviceaccount
- adds `--authfile` flag to `cosa push-container` invocations
- removes image tag param from baseos and extensions builds because the default is already in the format we want
- tweaks pipeline timeout to 4hrs
- tweaks tasks timeout to 1hr 30mins

LorbusChris commented:
@sherine-k thank you for the secret workspace suggestion!
It seems the Tekton way of doing credentials that we previously used is disabled: https://github.com/operate-first/apps/blob/3fe2abcbe5c829502760df79c5f01244c35fa50c/cluster-scope/base/operator.tekton.dev/tektonconfigs/config/tektonconfig.yaml#L29
That config must've not been applied until the `smaug` update/migration happened over the last couple of days, and only took effect now, which would explain why it did work at first and then broke for us.
